installer/install_delfin.py

```diff
@@ -53,42 +54,47 @@ def create_delfin_db():
 
 
 def start_processes():
-    # start api process
-    proc_path = os.path.join(delfin_source_path, 'delfin', 'cmd', 'api.py')
-    command = 'python3 ' + proc_path + ' --config-file ' +\
-              conf_file + ' >' + DEVNULL + ' 2>&1 &'
-    # >/dev/null 2>&1
-    logger.info("Executing command [%s]", command)
-    os.system(command)
-    logger.info("API process_started")
-
-    # Start task process
-    proc_path = os.path.join(delfin_source_path, 'delfin', 'cmd', 'task.py')
-    command = 'python3 ' + proc_path + ' --config-file ' +\
-              conf_file + ' >' + DEVNULL + ' 2>&1 &'
-    logger.info("Executing command [%s]", command)
-    os.system(command)
-
-    logger.info("TASK process_started")
-
-    # Start alert process
-    proc_path = os.path.join(delfin_source_path, 'delfin', 'cmd', 'alert.py')
-    command = 'python3 ' + proc_path + ' --config-file ' +\
-              conf_file + ' >' + DEVNULL + ' 2>&1 &'
-    logger.info("Executing command [%s]", command)
-    os.system(command)
-    logger.info("ALERT process_started")
+    processes = ['api', 'task', 'alert']
+    # Start cmd processes
+    for process in processes:
+        env_var = 'DELFIN_' + process.upper() + '_INSTANCES'
+        try:
+            instances = None
+            # Ignore multiple instance of api
+            if process != 'api':
+                instances = os.environ.get(env_var)
+            if not instances:
```

Comment:
do we need this? start_process arg default value is 1

Reply:
The reason the check is required because ```instances``` is type casted into ```int``` before passing to ```start_process```. otherwise it would give ```NoneType```

Reply:
Optimized the code a little bit based on suggestion